onnxruntime/core/providers/cuda/rnn/cudnn_rnn_base.cc

```diff
@@ -233,11 +234,11 @@ Status CudnnRnnBase<T>::ComputeInternal(OpKernelContext* ctx) const {
     y_alloc_data = GetScratchBuffer<T>(output_size);
     y_data = y_alloc_data.get();
   }
+  cudaMemset(y_data, 0, output_size * byte_size);
   const int32_t* sequence_lens_data = (sequence_lens == nullptr) ? nullptr : sequence_lens->template Data<int32_t>();
```

Comment:
Please add a comment saying why this is being done so the reason is captured with the code.

Reply:
added